google/cloud/grpc_error_delegate.cc

```diff
@@ -62,25 +63,34 @@ google::cloud::StatusCode MapStatusCode(grpc::StatusCode const& code) {
 }  // namespace
 
 google::cloud::Status MakeStatusFromRpcError(grpc::Status const& status) {
+  auto e = status.error_details();
+  if (!e.empty()) {
+    google::rpc::Status proto;
+    if (!proto.ParseFromString(e)) {
+      return google::cloud::Status(StatusCode::kInternal,
```

Comment:
Is that what we want?  We would be dropping the error message reported by the service in favor of the `kInternal`?  I think maybe you want to return a status with the original `status.error_code()` and `status.error_message()` and make the error details say "we got some error details but they were in an invalid format, so we had to discard them".

Reply:
Your suggestion is probably better. Changed. The reason I originally used `kInternal` was because this situation indicates a bad `grpc::Status` object, so it's unclear if its code and message are to be trusted either.